extensions/arc/src/ui/dashboards/postgres/postgresResourceHealthPage.ts

```diff
@@ -0,0 +1,343 @@
+
+		this.initialized = true;
+		return root;
+	}
+
+	protected get toolbarContainer(): azdata.ToolbarContainer {
+		// Refresh
+		const refreshButton = this.modelView.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: loc.refresh,
+			iconPath: IconPathHelper.refresh
+		}).component();
+
+		this.disposables.push(
+			refreshButton.onDidClick(async () => {
+				refreshButton.enabled = false;
+				try {
+					this.podConditionsLoading!.loading = true;
+
+					await Promise.all([
+						this._postgresModel.refresh(),
+						this._controllerModel.refresh()
+					]);
+				} catch (error) {
+					vscode.window.showErrorMessage(loc.refreshFailed(error));
+				}
+				finally {
+					refreshButton.enabled = true;
+				}
+			}));
+
+		return this.modelView.modelBuilder.toolbarContainer().withToolbarItems([
+			{ component: refreshButton, toolbarSeparatorAfter: true }
+		]).component();
+	}
+
+	private createPodList(): string[] {
+		const podStatus = this._postgresModel.config?.status.podsStatus;
+		let podNames: string[] = [];
+
+		podStatus?.forEach(p => {
+			let podHealthModels: PodHealthModel[] = [];
+			let indexes: number[] = [];
+
+
+			p.conditions.forEach(c => {
+				const condtionContainer = this.modelView.modelBuilder.flexContainer().withProps({
+					CSSStyles: { 'alignItems': 'center', 'height': '15px' }
+				}).component();
+
+				if (c.status === 'False') {
+					const imageComponent = this.modelView.modelBuilder.image().withProps({
+						iconPath: IconPathHelper.fail,
+						width: iconSize,
+						height: iconSize,
+						iconHeight: '15px',
+						iconWidth: '15px'
+					}).component();
+					condtionContainer.addItem(imageComponent, { CSSStyles: { 'margin-right': '0px' } });
+
+					condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+						value: c.message,
+					}).component());
+				} else {
+					const imageComponent = this.modelView.modelBuilder.image().withProps({
+						iconPath: IconPathHelper.success,
+						width: iconSize,
+						height: iconSize,
+						iconHeight: '15px',
+						iconWidth: '15px'
+					}).component();
+					condtionContainer.addItem(imageComponent, { CSSStyles: { 'margin-right': '0px' } });
+
+					if (c.type === 'Initialized') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.podInitialized,
+						}).component());
+					} else if (c.type === 'Ready') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.podReady,
+						}).component());
+					} else if (c.type === 'ContainersReady') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.containerReady,
+						}).component());
+					} else if (c.type === 'PodScheduled') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.podScheduled,
+						}).component());
+					}
```

Comment:
It might be good to do some future proofing here in case new conditions are added.  I'd add an else statement that adds `c.message` in case there's a new successful condition that wants to convey a message.